pyfstat/make_sfts.py

```diff
@@ -85,38 +87,24 @@ def __init__(
                 "Options 'minStartTime' and 'maxStartTime' are no longer supported!"
             )
 
+        if (noiseSFTs is None) and (SFTconstraint is not None):
+            raise ValueError("SFTconstraint can only be used with noiseSFTs")
+
         self.set_ephemeris_files()
         self.basic_setup()
         self.calculate_fmin_Band()
 
         self.tbounds = [self.tstart, self.tend]
         logging.info("Using segment boundaries {}".format(self.tbounds))
 
-    def basic_setup(self):
+    def _get_setup_from_input_data(self):
```

Comment:
Let's call this 
```suggestion
    def _get_setup_from_tstart_duration(self):
```
because until I was half-way through the function, I thought "data" meant files. ;)